Prometheus/Counter.cs

```diff
@@ -10,22 +10,30 @@ internal Child(Collector parent, LabelSequence instanceLabels, LabelSequence fla
             }
```

Comment:
The following comment from the go implementation has not yet been addressed, looks like encoding of the metric family header lines will also need to  format aware in this PR: 

```
// - Counters are expected to have the `_total` suffix in their metric name. In
//   the output, the suffix will be truncated from the `# TYPE` and `# HELP`
//   line. A counter with a missing `_total` suffix is not an error. However,
//   its type will be set to `unknown` in that case to avoid invalid OpenMetrics
//   output.
```